packages/esm-service-queues-app/src/add-provider-queue-room/add-provider-queue-room.component.tsx

```diff
@@ -33,110 +32,125 @@ import {
 } from '../helpers/helpers';
 import useQueueServices from '../hooks/useQueueService';
 import styles from './add-provider-queue-room.scss';
+import { z } from 'zod';
+import { Controller, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
 
 interface AddProviderQueueRoomProps {
   closeModal: () => void;
   providerUuid: string;
 }
 
+const createProviderQueueRoomSchema = (t: TFunction) =>
+  z.object({
+    queueLocationUuid: z.string({
+      required_error: t('missingQueueLocation', 'Please select a queue location'),
+    }),
+    queueProviderMapUuid: z.string(),
+    queueRoomUuid: z
+      .string({
+        required_error: t('missingQueueRoom', 'Please select a queue room'),
+      })
+      .min(1, t('missingQueueRoom', 'Please select a queue room')),
+    currentIsPermanentProviderQueueRoom: z.boolean().or(z.string()),
+  });
+
+type ProviderQueueRoomData = z.infer<ReturnType<typeof createProviderQueueRoomSchema>>;
+
 const AddProviderQueueRoom: React.FC<AddProviderQueueRoomProps> = ({ closeModal, providerUuid }) => {
   const { t } = useTranslation();
-  const { providerRoom } = useProvidersQueueRoom(providerUuid);
-  const currentIsPermanentProviderQueueRoom = useIsPermanentProviderQueueRoom() ?? false;
+  const { providerRoom, mutate } = useProvidersQueueRoom(providerUuid);
+  const isPermanentProviderQueueRoom = useIsPermanentProviderQueueRoom() ?? false;
   const currentLocationName = useSelectedQueueLocationName();
   const currentLocationUuid = useSelectedQueueLocationUuid();
   const currentService = useSelectedService();
 
-  const [isMissingQueueRoom, setIsMissingQueueRoom] = useState(false);
-  const [queueProviderMapUuid, setQueueProviderMapUuid] = useState('');
-  const [queueRoomUuid, setQueueRoomUuid] = useState('');
-
-  useEffect(() => {
-    if (providerRoom?.length > 0) {
-      setQueueRoomUuid(providerRoom?.[0].queueRoom?.uuid);
-      setQueueProviderMapUuid(providerRoom?.[0].uuid);
-    }
-  }, [providerRoom]);
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+    setValue,
+    watch,
+  } = useForm<ProviderQueueRoomData>({
+    mode: 'all',
+    resolver: zodResolver(createProviderQueueRoomSchema(t)),
+    defaultValues: {
+      queueLocationUuid: currentLocationUuid ?? '',
+      queueProviderMapUuid: providerRoom?.[0]?.uuid ?? '',
+      queueRoomUuid: providerRoom?.[0]?.queueRoom?.uuid ?? '',
+      currentIsPermanentProviderQueueRoom: isPermanentProviderQueueRoom,
```

Comment:
```suggestion
      currentIsPermanentProviderQueueRoom: isPermanentProviderQueueRoom ?? false
```

Maybe it's a good idea to a fallback default value for `currentIsPermanentProviderQueueRoom`.